core/src/main/java/org/elasticsearch/cluster/routing/allocation/command/AllocateEmptyPrimaryAllocationCommand.java

```diff
@@ -127,6 +129,10 @@ public RerouteExplanation execute(RoutingAllocation allocation, boolean explain)
                 shardRouting.unassignedInfo().getLastAllocationStatus());
         }
 
+        Logger logger = Loggers.getLogger(AllocateEmptyPrimaryAllocationCommand.class);
+        logger.warn("Allocating an empty primary for [{}][{}] on node [{}]. This action can cause data loss. If the " +
```

Comment:
I'm not sure if this is the right place for this, it logs independently of whether or not the cluster state change is successful, and even if this is being done as a dry run.